File: c/sumOfSquares.c
#include <stdio.h>

int sumOfSquares(int* arr)
{
    int sum = 0;
    for (int i = 0; i < sizeof(arr)/sizeof(int); i++)
        sum += arr[i] * arr[i];
    return sum;
}

int main()
{
    int* arr = {1, 2, 3, 4, 5};
    printf("%d", sumOfSquares(arr));
    return 0;
}

File: rust/sum_of_squares.rs
fn sum_of_squares(arr) {
    let sum = 0;
    for i in arr {
        sum += i;
    }
    return sum;
}

// driver code
fn main() {
    println("{}", sum_of_squares([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]));
}
